include/pybind11/pybind11.h

```diff
@@ -561,13 +571,34 @@ class cpp_function : public function {
                 // We'll need to copy this if we steal some kwargs for defaults
                 dict kwargs = reinterpret_borrow<dict>(kwargs_in);
 
+                // 1.5. Fill in any missing pos_only args from defaults if they exist
+                if (args_copied < func.nargs_pos_only) {
+                    for (; args_copied < func.nargs_pos_only; ++args_copied) {
+                        const auto &arg = func.args[args_copied];
+                        handle value;
+
+                        if (arg.value) {
+                            value = arg.value;
+                        }
+                        if (value) {
+                            call.args.push_back(value);
+                            call.args_convert.push_back(arg.convert);
+                        } else
+                            break;
```

Comment:
I haven't checked the rest of the codebase, so if this is consistent, leave it. I'm just a little bothered by `else` block not having braces while the `if` block does.

Reply:
It's done the same way just below, in the keyword checking.